src/Pim/Component/Catalog/spec/Validator/Constraints/UniqueValueValidatorSpec.php

```diff
@@ -44,4 +45,125 @@ function it_is_a_validator()
     {
         $this->shouldImplement(ConstraintValidatorInterface::class);
     }
+
+    function it_builds_a_violation_if_the_value_is_already_in_database_for_another_product(
+        ProductValueInterface $value,
+        UniqueValue $constraint,
+        AttributeInterface $releaseDate,
+        ProductInterface $product,
+        ConstraintViolationBuilderInterface $constraintViolationBuilder,
+        $context,
+        $uniqueDataRepository
+    ) {
+        $context->getRoot()->willReturn($product);
+        $releaseDate->isUnique()->willReturn(true);
+        $releaseDate->getCode()->willReturn('release_date');
+
+        $value->getAttribute()->willReturn($releaseDate);
+        $value->__toString()->willReturn('2015-16-03');
+
+        $uniqueDataRepository->uniqueDataNotExistsInThisProduct($value, $product)->willReturn(true);
```

Comment:
not what you check imho! you check only for the current ptoduct when it has an id (cf previous comment)